### PR TITLE
fix vue reactivity of rows by changing the reference of the updated row

### LINKED ISSUE #7938: [Telemetry Table] in place updates are not being rendered
#### Summary
VIPERGC-692
In place datum updates to existing rows are not being rendered in tables
Regression of #6694 

#### Expected vs Current Behavior
If new row data comes in, and the row exists, the new data should be displayed in the row.
Currently this does not happen

#### Steps to Reproduce
See VIPERGC-692

#### Testing Instructions
For full production testing, see VIPERGC-692.
Local reproduction steps
1. Create an `Event Message Generator with Acknowledge` object
  2. if you don't see this option, open the browser console and enter `openmct.install(openmct.plugins.example.EventGeneratorPlugin())`
3. Change the time conductor to realtime mode
4. Observe after 5 seconds an unacknowledged row will come in and be added to the table
5. Observe after 5 seconds the row's `Acknowledge` column will be updated with `OK`.

#### Environment
* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by davetsay**
Closes #7938 

### Describe your changes:
* make in place row updates trigger vue reactivity
* do not trigger row update calculations on horizontal scroll

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/telemetryTable/collections/TableRowCollection.js [hunk 1]
```diff
@@ -124,10 +129,22 @@ export default class TableRowCollection extends EventEmitter {
     return foundIndex;
   }
 
-  updateRowInPlace(row, index) {
-    const foundRow = this.rows[index];
-    foundRow.updateWithDatum(row.datum);
-    this.rows[index] = foundRow;
```
Comment (akhenry) [inline src/plugins/telemetryTable/collections/TableRowCollection.js:130]: 🤦 _this_ is [why reactivity was broken](https://vuejs.org/guide/essentials/list.html#mutation-methods). Good catch!

**Review by akhenry — CHANGES_REQUESTED** [review]
Just wondering whether the `splice` change by itself fixes it?

Also, can you explain the scrollX issue? Is it related to in-place row updates? The scope of change approved by the CCB is for fixing in-place row updates.

**Comment by davetsay** [comment]
> Just wondering whether the `splice` change by itself fixes it?
> 
> Also, can you explain the scrollX issue? Is it related to in-place row updates? The scope of change approved by the CCB is for fixing in-place row updates.

The incoming vs existing is more slimming the code down. We create new `TelemetryTableRow`s, so their datum is already normalized, etc, so we don't have to re-perform that operation. So, updateWithDatum being slimmed down is just a code improvement. Splice vs index replacement is not entirely it. You can index replace with a reference to a new row to trigger reactivity.

The horizontal scroll I came across when I started wondering if throttled calls was affecting adding rows and updating rows. We are calling `updateView` and recalculating `visibleRows` on every vertical AND horizontal scroll. It's unnecessary on horizontal scroll, but can remove if you feel it introduces too much risk.

**Review by akhenry — APPROVED** [review]

File: src/plugins/telemetryTable/collections/TableRowCollection.js [hunk 2]
```diff
@@ -124,10 +129,22 @@ export default class TableRowCollection extends EventEmitter {
...
+   * so merge existing and incoming row properties
+   *
+   * Do to reactivity of Vue, we want to replace the existing row with the updated row
+   * @param {TelemetryTableRow} incomingRow to update
+   * @param {number} index of the existing row in the collection to update
+   */
+  updateRowInPlace(incomingRow, index) {
+    // Update the incoming row, not the existing row
+    const existingRow = this.rows[index];
+    incomingRow.updateWithDatum(existingRow);
```
Comment (RadotJar) [inline src/plugins/telemetryTable/collections/TableRowCollection.js:143]: Shouldn't this be the other way around? As far as I understand, the intent of the code is to update the existingRow with the new data in incomingRow. This looks to me to do the opposite: Preserving as much of existingRow as possible.

**Pull request merged**